nodewatcher/modules/monitor/http/survey/processors.py

```diff
@@ -0,0 +1,106 @@
+
+
+class SurveyInfo(monitor_processors.NodeProcessor):
+    """
+    Accesses the data collected during a WiFi survey of neighboring wireless access points / routers. Stores channel,
+    BSSID, SSID and signal strength for each neighbor into the datastream. A star shaped graph is constructed. The
+    source vertex contains an array of all BSSIDs of that node. Source vertex is stored according to its UUID,
+    wherease all neighbors are stored according to their BSSIDs. Will only run if HTTP monitor module has previously
+    fetched data.
+    """
+
+    @monitor_processors.depends_on_context("http", http_processors.HTTPTelemetryContext)
+    def process(self, context, node):
+        """
+        Called for every processed node.
+
+        :param context: Current context
+        :param node: Node that is being processed
+        :return: A (possibly) modified context
+        """
+
+        version = context.http.get_module_version('core.wireless')
+        if version == 0:
+            # Unsupported version or data fetch failed
+            return context
+
+        edges, vertices = [], []
+        try:
+            source_vertex_bssids = [interface.bssid for interface in context.http.core.wireless.interfaces.values()]
+            source_vertex = {
+                'i': str(node.uuid),
+                'bssids': source_vertex_bssids,
+            }
+            vertices.append(source_vertex)
+            for radio in context.http.core.wireless.radios.values():
+                for neighbor in radio.survey:
+                    vertices.append({'i': neighbor['bssid']})
+                    # Add an edge for that neighbor.
+                    edge = {'f': str(node.uuid), 't': neighbor['bssid']}
+                    for field in ('channel', 'signal', 'ssid'):
+                        if field in neighbor:
+                            edge[field] = neighbor[field]
+                    edges.append(edge)
+        except KeyError:
+            pass
+        except NameError:
+            self.logger.warning("Could not parse survey data for node: " + str(node))
+
+        latest_stored_graph = None
+        # Retrieve the latest stored datapoint, if it exists
+        streams = datastream.find_streams({'node': node.uuid})
+        if streams:
+            latest_stream = streams[len(streams) - 1]
+            datapoint_iterator = datastream.get_data(stream_id=latest_stream['stream_id'],
+                                                     granularity=latest_stream['highest_granularity'],
+                                                     start=latest_stream['latest_datapoint'])
+            try:
+                latest_stored_graph = datapoint_iterator[0]['v']
+            except IndexError:
+                pass
+
+        latest_graph = {
+            "v": vertices,
+            "e": edges
+        }
+
+        if latest_graph != latest_stored_graph:
+            # Store the latest graph into datastream.
```

Comment:
> So stored timestamp should represent the same time as your local time timestamp, only in UTC timezone.

This does happen, so I think it's OK. I was wondering why you'd want to store values into the database according to the UTC timezone instead of the one specified in the settings.